### PR TITLE
net-p2p/amule: fix build w/ clang and boost-1.82

### PR DESCRIPTION
**Pull request by puleglot**
Closes: https://bugs.gentoo.org/907843

### CODE REVIEW

File: net-p2p/amule/files/amule-2.3.3-boost-m4.patch [hunk 1]
```diff
@@ -0,0 +1,968 @@
+Bug: https://bugs.gentoo.org/907843
+Source: https://github.com/tsuna/boost.m4/raw/master/build-aux/boost.m4
```
Comment (thesamesam) [inline net-p2p/amule/files/amule-2.3.3-boost-m4.patch:2]: Can we BDEPEND on sys-devel/boost-m4 (possibly bumping it first) and copy the macro in from there? It'd avoid needing a huge patch.

Comment (puleglot) [reply net-p2p/amule/files/amule-2.3.3-boost-m4.patch:2]: Sure. I didn't know it existed.

> and copy the macro in from there

I think just removing it from the source dir is enough.

Comment (thesamesam) [reply net-p2p/amule/files/amule-2.3.3-boost-m4.patch:2]: Done, thanks.

**Comment by puleglot** [comment]
I've dropped `~riscv` keyword for now because `sys-devel/boost-m4` is missing it and appended -std=gnu++14 for compatibility with clang >=16

**Pull request closed without merging**